python/lsst/ip/isr/calibType.py

```diff
@@ -261,11 +270,44 @@ def updateMetadata(self, camera=None, detector=None, filterName=None,
         self._metadata["DET_SER"] = self._detectorSerial if self._detectorSerial else None
         self._metadata["FILTER"] = self._filter if self._filter else None
         self._metadata["CALIB_ID"] = self._calibId if self._calibId else None
+        self._metadata["SEQFILE"] = self._seqfile if self._seqfile else None
+        self._metadata["SEQNAME"] = self._seqname if self._seqname else None
+        self._metadata["SEQCKSUM"] = self._seqcksum if self._seqcksum else None
         self._metadata["CALIBCLS"] = get_full_type_name(self)
 
         mdSupplemental.update(kwargs)
         mdOriginal.update(mdSupplemental)
 
+    def updateMetadataFromExposures(self, exposures):
+        """Extract and unify metadata information.
+
+        Parameters
+        ----------
+        exposures : `list`
+            Exposures or other calibrations to scan.
+        """
+        # This list of keywords is the set of header entries that
+        # should be checked and propagated.  Not having an entry is
+        # not a failure, as they may not be defined for the exposures
+        # being used.
+        keywords = ["SEQNAME", "SEQFILE", "SEQCKSUM", "ODP", "AP0_RC"]
+        metadata = {}
+
+        for exp in exposures:
+            try:
+                expMeta = exp.getMetadata()
+            except AttributeError:
+                continue
+            for key in keywords:
+                if key in expMeta:
+                    if key in metadata:
+                        if metadata[key] != expMeta[key]:
+                            self.log.warning("Metadata mismatch! Have: %s Found %s",
+                                             metadata[key], expMeta[key])
```

Comment:
_Kinda_ wonder if this should raise... This is lossy, as we just end up taking the first value, so if you don't spot this warning you lose the info, right? We could make the item a list item in these cases, and yes, that would cause problems downstream, but that would be half the point - to let you know you have mixed inputs to a calib. The latter was just a passing thought and probably not a good idea, but the former point remains - should this raise or warn?

Reply:
I've been saving and skimming the calib construction log files, so I left this as a warning as it'd be something I'd look for.  There are also situations where I didn't think a raise was appropriate (crosstalk was my first thought, and I think flats would be another as the flux levels should swamp out any voltage related few-ADU differences).

Reply:
Ah, OK, I see, thanks.